internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -119,6 +119,10 @@ func (m *DynamicScaleUpManager) processDynamicScaleUpReconcileEvent() error {
 		}
 	}
 
+	if len(errList) > 0 {
```

Comment:
There is no need to add this condition here. I think we can just return `errList`

Reply:
This is needed because otherwise the lint ended up complaining that a non nil value is never returned.

Reply:
```
internal/kafka/internal/workers/cluster_mgrs/cleanup_cluster_external_resources_mgr.go:94:5: SA4023: this comparison is always true (staticcheck)
	if err != nil {
	   ^
internal/kafka/internal/workers/cluster_mgrs/cleanup_cluster_external_resources_mgr.go:93:9: SA4023(related information): the lhs of the comparison is the 1st return value of this function call (staticcheck)
	err := c.processCleanupClusters()
	       ^
internal/kafka/internal/workers/cluster_mgrs/cleanup_cluster_external_resources_mgr.go:101:50: SA4023(related information): (*github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/workers/cluster_mgrs.CleanupClusterExternalResourcesManager).processCleanupClusters never returns a nil interface value (staticcheck)
func (c *CleanupClusterExternalResourcesManager) processCleanupClusters() error {
```

it was complaining with something like this..

Reply:
:+1:

Reply:
I think the issue here is that
	var errList fleeterrors.ErrorList

is `nil`,

but when converted to an error it becomes an interface that is not nil, but it contains a nil value

Reply:
I don't think that is the correct interpretation. The issue here, IMO, is that ErrList is not a pointer, so it can't be `nil`.
Internally it is declared as a slice, so ErrorList, in the beginning, is an empty slice.

You can see try it yourself here:
https://go.dev/play/p/94SP5EqWyE_z

Reply:
`ErrList` is a pointer and it can be nil.
See https://go.dev/play/p/Gbs3mrR7_tJ

Reply:
Never said it can't be nil, just it would be a long discussion to explain it here. I could make a blog for that :)
Have you looked at my code? That is **exactly** as the code in this PR and it is an empty array.

Reply:
I looked at your code yes, and if I understand the behavior properly what I said still applies.

The confusion you are having is that Println doesn't show the difference between empty and nil in the case of slices. It prints both cases as `[]`.

Add the following code at the end of your example:
```
	fmt.Println(reflect.ValueOf(err).IsNil())
```

and you will be able to see that

Reply:
You are totally right. Sorry for the confusion. Probably my holiday was not enough 😄

Reply:
:-) Thanks for the discussion @ziccardi @miguelsorianod 

Just adding in my2cents: I didn't dig further the reason why but I believe the debugger was showing the similar behavior described in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1266#discussion_r957177361